packages/fxa-auth-server/test/local/payments/configuration/manager.js

```diff
@@ -0,0 +1,260 @@
+}
+
+async function deleteCollection(db, collectionRef, batchSize) {
+  const query = collectionRef.orderBy('__name__').limit(batchSize);
+
+  return new Promise((resolve, reject) => {
+    deleteQueryBatch(db, query, resolve).catch(reject);
+  });
+}
+
+describe('PaymentConfigManager', () => {
+  let paymentConfigManager;
+  let testProductId;
+  let testPlanId;
+  let productConfigDbRef;
+  let planConfigDbRef;
+
+  beforeEach(async () => {
+    testProductId = randomUUID();
+    testPlanId = randomUUID();
+    const firestore = setupFirestore(mockConfig);
+    Container.set(AuthFirestore, firestore);
+    Container.set(AuthLogger, {});
+    Container.set(AppConfig, mockConfig);
+    paymentConfigManager = new PaymentConfigManager();
+    productConfigDbRef = paymentConfigManager.productConfigDbRef;
+    planConfigDbRef = paymentConfigManager.planConfigDbRef;
+    await productConfigDbRef.doc(testProductId).set(productConfig);
+    await planConfigDbRef
+      .doc(testPlanId)
+      .set({ ...planConfig, productId: 'test-product' });
+  });
+
+  afterEach(async () => {
+    const productConfigDbRef = paymentConfigManager.productConfigDbRef;
+    const planConfigDbRef = paymentConfigManager.planConfigDbRef;
+    await deleteCollection(
+      paymentConfigManager.firestore,
+      productConfigDbRef,
+      100
+    );
+    await deleteCollection(
+      paymentConfigManager.firestore,
+      planConfigDbRef,
+      100
+    );
+    Container.reset();
+  });
+
+  describe('load', async () => {
+    it('loads products and plans and returns them', async () => {
+      await paymentConfigManager.load();
+      const products = paymentConfigManager.allProducts();
+      const plans = paymentConfigManager.allPlans();
+      assert.equal(products.length, 1);
+      assert.equal(plans.length, 1);
+    });
+  });
+
+  describe('listeners', () => {
+    it('starts and stops', async () => {
+      await paymentConfigManager.startListener();
+      await paymentConfigManager.stopListener();
+    });
+
+    it('registers new/updates plans and products', async () => {
+      const newProduct = cloneDeep(productConfig);
+      newProduct.id = randomUUID();
+      const newPlan = cloneDeep(planConfig);
+      newPlan.id = randomUUID();
+      newPlan.productId = newProduct.id;
+      await paymentConfigManager.load();
+
+      await paymentConfigManager.startListener();
+      let products = paymentConfigManager.allProducts();
+      let plans = paymentConfigManager.allPlans();
+      assert.equal(products.length, 1);
+      assert.equal(plans.length, 1);
+
+      // Insert a new product/plan
+      await productConfigDbRef.doc(newProduct.id).set(productConfig);
+      await planConfigDbRef.doc(newPlan.id).set(newPlan);
+
+      // Because this may take a variable amount of time and to avoid
+      // test flakiness on different machines, we retry until we get the
+      // expected number of products/plans within a reasonable time.
+      await retry(
```

Comment:
I couldn't find how we resolved flaky tests elsewhere in auth-server (maybe @vbudhram knows), but when waiting for something that should happen 'relatively quick', this approach seems pretty reliable and easy, which sets a lower and upper bound for how long the async thing may take.

Reply:
@bbangert In the auth-server we have typically increase the test timeout on a case by case situation and for other services we have added a small `Promise.delay` as needed. Using `async-retry` seems like a better approach here.